core/bindings.rs

```diff
@@ -140,6 +140,26 @@ pub fn set_func_raw(
   obj.set(scope, key.into(), val.into());
 }
 
+pub extern "C" fn wasm_async_resolve_promise_callback(
+  isolate: *mut v8::Isolate,
+  context: v8::Local<v8::Context>,
+  resolver: v8::Local<v8::PromiseResolver>,
+  compilation_result: v8::Local<v8::Value>,
+  success: v8::WasmAsyncSuccess,
+) {
+  // SAFETY: This pointer is definitely valid for the lifetime of this function.
+  let isolate = unsafe { &mut *isolate };
+  isolate.set_microtasks_policy(v8::MicrotasksPolicy::Explicit);
```

Comment:
This callback is called whenever a wasm streaming resource is closed or aborted, which happens from Deno-internal JS code (the `handleWasmStreaming` function in `26_fetch.js`). Therefore, even if we skip the microtask checkpoint when resolving `resolver`, the microtasks will run when the JS stack is emptied, which is just after this callback runs. As such, setting the microtasks policy doesn't actually do anything – it's setting the callback that makes the difference.

Reply:
Thanks, I removed that. I will check later why the callback needs to be set.